core/cthreads.c

```diff
@@ -0,0 +1,291 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "cthreads.h"
+
+#if __unix__
+#include <pthread.h>
+void *__cthreads_pthread_function_wrapper(void *data) {
+  struct cthreads_args *args = data;
+  args->func(args->data);
+
+  free(data);
+
+  return NULL;
+}
+#elif _WIN32
+#include <windows.h>
+DWORD WINAPI __cthreads_winthreads_function_wrapper(void *data) {
+  struct thread_args *args = data;
+  args.func(args.data);
+
+  free(data);
+
+  return TRUE;
+}
+#endif
+
+int cthreads_thread_create(struct cthreads_thread *thread, void *(*func)(void *data), void *data) {
+  #ifdef __unix__
+    pthread_t pthread;
+
+    struct cthreads_args *args = malloc(sizeof(struct cthreads_args));
+    args->func = func;
+    args->data = data;
+
+    int res = pthread_create(&pthread, NULL, __cthreads_pthread_function_wrapper, (void *)args);
```

Comment:
Would be nice if you manage to keep the attributes param.. (for both unix and windows)